base/game_events/PsynergyStoneEvent.ts

```diff
@@ -0,0 +1,154 @@
+            const event = this.data.game_event_manager.get_event_instance(event_info);
+            this.finish_events.push(event);
+        });
+    }
+
+    next() {
+        this.control_enable = false;
+        this.dialog_manager.next(finished => {
+            this.control_enable = true;
+            if (finished) {
+                this.resolve();
+            }
+        });
+    }
+
+    async _fire(origin_npc?: NPC) {
+        if (!this.active) return;
+        this.origin_npc = origin_npc;
+        ++this.data.game_event_manager.events_running_count;
+        this.control_enable = false;
+        this.running = true;
+
+        //Initial Text
+        const hero_name = this.data.info.party_data.members[0].name;
+        await this.set_text(this.custom_init_text ?? INIT_TEXT(hero_name));
+
+        await this.data.hero.face_direction(directions.down);
+        this.data.game_event_manager.force_idle_action = false;
+        this.data.hero.play(base_actions.GRANT);
+        this.data.audio.pause_bgm();
+        this.data.audio.play_se("misc/item_get", () => {
+            this.data.audio.resume_bgm();
+        });
+
+        //Stone final position
+        const stone_sprite_x = this.data.hero.sprite.centerX;
+        const stone_sprite_y = this.data.hero.sprite.centerY - 24;
+
+        //Particle emitter
+        const emitter = this.game.add.emitter(stone_sprite_x, stone_sprite_y, 15);
+        const psynergy_particle_base = this.data.info.misc_sprite_base_list["psynergy_particle"];
+        const sprite_key = psynergy_particle_base.getSpriteKey("psynergy_particle");
+        emitter.makeParticles(sprite_key);
+        emitter.minParticleSpeed.setTo(-1, -1);
+        emitter.maxParticleSpeed.setTo(1, 1);
+        emitter.gravity = 100;
+        emitter.width = emitter.height = 20;
+        emitter.forEach((particle: Phaser.Sprite) => {
+            psynergy_particle_base.setAnimation(particle, "psynergy_particle");
+        });
+
+        emitter.start(false, 300, 250, 0);
+        const anim_key = psynergy_particle_base.getAnimationKey("psynergy_particle", "vanish");
+        emitter.forEach((particle: Phaser.Sprite) => {
+            particle.animations.play(anim_key, 10);
+            particle.animations.currentAnim.setFrame((Math.random() * particle.animations.frameTotal) | 0);
+        });
+
+        this.game.add
+            .tween(this.origin_npc.sprite.body)
+            .to({x: stone_sprite_x, y: stone_sprite_y}, 300, Phaser.Easing.Linear.None, true);
+        await this.set_text(GET_TEXT(hero_name));
+
+        //Recover PP and show status window
+        this.chars_status_window.show();
+        this.stone_pp_recovery();
+        this.chars_status_window.update_chars_info();
+        this.chars_status_window.update_position();
+        this.data.audio.play_se("battle/heal_1");
+        await this.set_text(RECOVERY_TEXT(hero_name));
+
+        emitter.destroy();
+        let closed: () => void;
```

Comment:
Don't need to create this var. The callback arg in `close` function is optional.